pandas/core/groupby/groupby.py

```diff
@@ -2507,6 +2543,7 @@ def shift(self, periods=1, freq=None, axis=0, fill_value=None):
 
         return self._get_cythonized_result(
             "group_shift_indexer",
+            numeric_only=False,
```

Comment:
What did this need to change for?

Reply:
I added numeric_only for other aggregation functions (e.g. var), and defaulted it to True, the same as _cython_agg_general. Setting this to False gives the previous behavior.